client/src/components/ZoomableSVG.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { Component, useRef, useState, useEffect } from 'react';
+import * as d3 from 'd3';
+
+export default function ZoomableSVG( { children, width, height } ){
+
+    const svgRef = useRef()
+    const [k, setK] = useState(1)
```

Comment:
The name K is not meaningful. X and y can maybe be improved, but they are acceptable.

K should be renamed to scale or something similar.